rocky/tools/forms/ooi_form.py

```diff
@@ -109,25 +106,35 @@ def generate_select_ooi_field(
     field: FieldInfo,
     related_ooi_type: type[OOI],
     initial: str | None = None,
-) -> forms.fields.Field:
+) -> forms.fields.Field | None:
     # field is a relation, query all objects, and build select
     default_attrs = default_field_options(name, field)
     is_multiselect = getattr(field.annotation, "__origin__", None) is list
     option_label = default_attrs.get("label", _("option"))
 
-    option_text = "-- " + _("Optionally choose a {option_label}").format(option_label=option_label) + " --"
+    manytext = "one or more:" if is_multiselect else "a"
     if field.is_required():
-        option_text = "-- " + _("Please choose a {option_label}").format(option_label=option_label) + " --"
+        option_text = _("Please choose {manytext} {option_label}").format(option_label=option_label, manytext=manytext)
+    else:
+        option_text = _("Optionally choose {manytext} {option_label}").format(
+            option_label=option_label, manytext=manytext
+        )
 
     # Generate select options
-    select_options = [] if is_multiselect else [("", option_text)]
-
+    select_options = [("", "-- " + option_text + " --")]
     if initial:
         select_options.append((initial, initial))
 
     oois = api_connector.list_objects({related_ooi_type}, datetime.now(timezone.utc)).items
     select_options.extend([(ooi.primary_key, ooi.primary_key) for ooi in oois])
 
+    # Remove the selection option when only 1 OOI is available, and required
+    if field.is_required() and not initial and len(oois) == 1:
+        del select_options[0]
+    # Don't show select fields without options if they are not required
+    elif not field.is_required() and not len(oois):
```

Comment:
```suggestion
    elif not field.is_required() and not oois:
```

While looking at this I was also wondering:
- It's a bit odd that this is the only generator function that could yield a `None` instead of a field, usually you want to decide wether or not te render this field *outside* of this function
- Do we really need to skip rendering this field when there's nothing to choose? Imo displaying it with e.g. "no options available" or making it unselectable is a more consistent approach

Reply:
Since the user cannot select a value, and the field is optional my gut feeling says its a wrong move to ask the user to make a choice. Doing the detection of an empty field outside this function in in the loop that handles all the fields would add the need for another communication channel in which we tell the outer function about the selection choice length.

Reply:
@HeleenSG @RomijnHumanoids what do you guys think?

Reply:
What are the specific fields that are hidden or shown in pre-filled select with one option?

It depends on the context I would say, what users expect of the content. 
- If a network field is not expected in the form of a certain ooi type and the subject field only adds confusion to the form; it would beter to leave it out after that ooi type is selected. 
- If users expect the field to be there, maybe even expect to be able to make a selection and for some reason that is not possible in this form; it would be better to show the pre-filled single option field, or read-only state. If read only, possibly explain why different value is not possible.

I find it hard to judge the context of this specific content, maybe @stephanie0x00 could help interpret what’s expected in which situation?

Reply:
Fields that are not required, and have no selectable options will be hidden with this change. It depends on the ObjectType, and the current database contents. When the database has more selectable types, it will more often show these optional fields, whereas for a beginner/novice user their database is till empty (probably) and the questions they cannot answer (on that page) are no longer asked.